crates/core/src/connection.rs

```diff
@@ -103,6 +103,19 @@ impl Connection {
         }
     }
 
+    /// Begin a new transaction in DEFERRED mode, which is the default.
+    pub fn transaction(&self) -> Result<Transaction> {
+        self.transaction_with_behavior(TransactionBehavior::Deferred)
+    }
+
+    /// Begin a new transaction in the given mode.
+    pub fn transaction_with_behavior(
```

Comment:
imho I would add a version of this fn for each enum variant and then make the enum private. Its 1) cleaner and 2) more backwards compat w/ adding new variant (which I assume in this case is likely never to happen).

Reply:
This is just mirroring the rusqlite API.